File: sirius-ruby-client-win32/lib/sirius/win32/classes/controls/Slider.rb
module Sirius
  module Client
    module Win32
      module Classes
        module Controls
          class Slider < Control
            def initialize(locator, parent = nil)
              super(locator, parent)
            end

            # SYNOPSIS
            #   getLowerBound(parameters)
            #
            # ARGS
            #   parameters      GetLowerBound - {http://controls.classes.win32.server.sirius.org/}getLowerBound
            #
            # RETURNS
            #   parameters      GetLowerBoundResponse - {http://controls.classes.win32.server.sirius.org/}getLowerBoundResponse
            #
            def get_lower_bound
              @client.core.slider.getLowerBound(parameters)
            end

            # SYNOPSIS
            #   getUpperBound(parameters)
            #
            # ARGS
            #   parameters      GetUpperBound - {http://controls.classes.win32.server.sirius.org/}getUpperBound
            #
            # RETURNS
            #   parameters      GetUpperBoundResponse - {http://controls.classes.win32.server.sirius.org/}getUpperBoundResponse
            #
            def get_upper_bound
              @client.core.slider.getUpperBound(parameters)
            end

            # SYNOPSIS
            #   getPosition(parameters)
            #
            # ARGS
            #   parameters      GetPosition - {http://controls.classes.win32.server.sirius.org/}getPosition
            #
            # RETURNS
            #   parameters      GetPositionResponse - {http://controls.classes.win32.server.sirius.org/}getPositionResponse
            #
            def get_position
              @client.core.slider.getPosition(parameters)
            end

            # SYNOPSIS
            #   setPosition(parameters)
            #
            # ARGS
            #   parameters      SetPosition - {http://controls.classes.win32.server.sirius.org/}setPosition
            #
            # RETURNS
            #   parameters      SetPositionResponse - {http://controls.classes.win32.server.sirius.org/}setPositionResponse
            #
            def set_position
              @client.core.slider.setPosition(parameters)
            end
          end
        end
      end
    end
  end
end

File: sirius.server.win32/src/main/java/org/sirius/server/win32/classes/controls/Slider.java
/**
 * 
 */
package org.sirius.server.win32.classes.controls;

import java.rmi.RemoteException;

import javax.jws.WebService;

import org.sirius.server.win32.classes.Common;
import org.sirius.server.win32lib.controls.Win32LibControlsClient;
import org.sirius.server.win32lib.controls.slider.ISliderContractProxy;

/**
 * @author Myk Kolisnyk
 *
 */
@WebService
public class Slider extends Common {
    
    private final Win32LibControlsClient win32lib;  
    
    /**
     * 
     */
    public Slider() {
        win32lib = new Win32LibControlsClient();
    }

    /**
     * @param hwnd
     * @return
     * @throws RemoteException
     * @see org.sirius.server.win32lib.controls.slider.ISliderContractProxy#getLowerBound(java.lang.Integer)
     */
    public Double getLowerBound(Integer hwnd) throws RemoteException {
        return win32lib.slider().getLowerBound(hwnd);
    }

    /**
     * @param hwnd
     * @return
     * @throws RemoteException
     * @see org.sirius.server.win32lib.controls.slider.ISliderContractProxy#getPosition(java.lang.Integer)
     */
    public Double getPosition(Integer hwnd) throws RemoteException {
        return win32lib.slider().getPosition(hwnd);
    }

    /**
     * @param hwnd
     * @return
     * @throws RemoteException
     * @see org.sirius.server.win32lib.controls.slider.ISliderContractProxy#getUpperBound(java.lang.Integer)
     */
    public Double getUpperBound(Integer hwnd) throws RemoteException {
        return win32lib.slider().getUpperBound(hwnd);
    }

    /**
     * @param hwnd
     * @param pos
     * @throws RemoteException
     * @see org.sirius.server.win32lib.controls.slider.ISliderContractProxy#setPosition(java.lang.Integer, java.lang.Double)
     */
    public void setPosition(Integer hwnd, Double pos) throws RemoteException {
        win32lib.slider().setPosition(hwnd, pos);
    }

}
